src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs

```diff
@@ -47,15 +51,15 @@ private bool HasReadOnlyFlag
             {
                 Debug.Assert(_state != Uninitialized); // Use this after EnsureCachesInitialized has been called.
 
-                if (!EntryExists)
+                if (!EntryExists || IsBrokenLink)
                 {
                     return false;
                 }
 
 #if TARGET_BROWSER
-                var mode = (Interop.Sys.Permissions)(_fileCache.Mode & (int)Interop.Sys.Permissions.Mask);
-                bool isUserReadOnly = (mode & Interop.Sys.Permissions.S_IRUSR) != 0 && // has read permission
-                                      (mode & Interop.Sys.Permissions.S_IWUSR) == 0;   // but not write permission
+                var mode = (UnixFileMode)(_fileCache.Mode & (int)FileSystem.ValidUnixFileModes);
```

Comment:
```suggestion
                var mode = ((UnixFileMode)_fileCache.Mode & FileSystem.ValidUnixFileModes);
```

So you only need 1 cast.